### ISSUE TITLE
Nullable: support `var?`

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#31874]
We need to discuss in LDM whether we want this feature, and if so, when. Also, there are open questions on its semantics.

Along those lines, would we allow `var? (x, y) = ...` deconstructions?

**Comment by Spongman** [comment]
IMHO this is a big hole in the language.

two examples i came across porting my codebase to c#8 with non-nullable checks:

```
var foo = new {};
foo = null
```


```
var foo = Foo(); // returns `Some<Really, IComplicated<Generic, Func<Type>>[], ugh>`
foo = null;
```

**Comment by pentp** [comment]
`var?` would be really helpful for out parameters (currently you have to spell out the type).

**Comment by jcouv** [comment]
This was discussed in LDM 2019/12/18: https://github.com/dotnet/csharplang/blob/master/meetings/2019/LDM-2019-12-18.md#var
It was rejected in favor of inferring a nullable type for `var` locals. 
Fix is upcoming

**Comment by jcouv** [comment]
Fix was merged to 16.6p1. Closing this issue now.
